optimum/onnxruntime/utils.py

```diff
@@ -187,6 +187,11 @@ def validate_provider_availability(provider: str):
                     raise ImportError(
                         f"`onnxruntime-gpu` is installed, but GPU dependencies are not loaded. It is likely there is a conflicting install between `onnxruntime` and `onnxruntime-gpu`. Please install only `onnxruntime-gpu` in order to use {provider}."
                     )
+                elif os.path.isfile(path_cuda_lib) and is_onnxruntime_training_available():
+                    if provider == "TensorrtExecutionProvider":
+                        raise ImportError(
+                            f"Asked to use {provider}, but `onnxruntime-training` package doesn't support {provider}. Please use `CUDAExecutionProvider` instead."
+                        )
```

Comment:
Are `onnxruntime-gpu` and `onnxruntime-training` incompatible? Somebody may have `onnxruntime-training` installed but still want to use TRT execution provider for inference.

Reply:
TRT ep is not available with `onnxruntime-training`.

Reply:
Are onnxruntime-gpu and onnxruntime-training incompatible?

Reply:
Yes.

Reply:
All good then